website/docs/internals/json-format.html.md

```diff
@@ -16,7 +16,15 @@ Since the format of plan files isn't suited for use with external tools (and lik
 
 Use `terraform show -json <FILE>` to generate a JSON representation of a plan or state file. See [the `terraform show` documentation](/docs/cli/commands/show.html) for more details.
 
--> **Note:** The output includes a `format_version` key, which currently has major version zero to indicate that the format is experimental and subject to change. A future version will assign a non-zero major version and make stronger promises about compatibility. We do not anticipate any significant breaking changes to the format before its first major version, however.
+The output includes a `format_version` key, which as of Terraform 1.1.0 has
+value `"1.0"`. The semantics of this version are:
+
+- Backwards compatible changes or additions will be made along with a minor
+  version increment, e.g. `"1.1"`. Consumers of this format should ignore any
+  unknown keys to remain forwards compatible.
+- Any backwards incompatible changes will result in a major version increment,
+  e.g. `"2.0"`. Consumers of this format should reject unknown major format
+  versions.
```

Comment:
I dunno if it necessarily needs to be called out here, but a literal read of this makes it seem in conflict with the Terraform v1.0 Compatibility Promises just because it doesn't say anything about what circumstances we might allow such a change.

Perhaps a reasonable compromise here would be to just add an additional sentence here which connects the two, to make it clear that we're not intending to make an exception here:

> - Any backwards incompatible changes will result in a major version increment, e.g. `"2.0"`. Consumers of this format should reject unknown major format versions. We will introduce new major versions only within the bounds of [the Terraform 1.0 Compatibility Promises](https://www.terraform.io/docs/language/v1-compatibility-promises.html).

(My intended implication of this reference, then, would be: a major version change to these formats can come only either as part of a future Terraform v2.0 or in response to some explicit opt-in which retains the 1.x format as the default result.)